app/controllers/orders_controller.rb

```diff
@@ -8,6 +8,19 @@ def create
     end
   end
 
+  def update
+    if user_signed_in?
+      @order = Order.find(params[:id])
+      if @order.nil?
+        render json: {message: "Cannot find order"}, status: :not_found
+      end
+      @order.update(order_params)
+      render json: {message: "User updated"}
```

Comment:
You are assuming that the `.update` function will go through, but you should include a if statement to check.